Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -2117,11 +2138,40 @@ public class TimeSlider: UIControl {
         return nil
     }
     
+    // This function handles change in operationalLayers
+    // and re-initialize time slider if required
+    private func handleOperationalLayers(change: NSKeyValueObservedChange<NSMutableArray>) {
+        //
+        // Make sure change contains time aware layer
+        guard let geoView = geoView, changeContainsTimeAwareLayer(change: change) else {
+            return
+        }
+        
+        // Re initialize time slider
+        reInitializeTimeProperties = true
+        initializeTimeProperties(geoView: geoView, observeGeoView: observeGeoView, completion: { [weak self] (error) in
+            //
+            // Bail out if there is an error
+            guard error == nil else {
+                return
+            }
+
+            // Make sure self is around
+            guard let strongSelf = self else {
+                return
+            }
+            
+            // Set the flag
+            strongSelf.reInitializeTimeProperties = false
+        })
+    }
+    
     // This function checks whether the observed value of operationalLayers
     // contains any time aware layer.
-    private func changeContainsTimeAwareLayer(change: [NSKeyValueChangeKey : Any]?) -> Bool {
-        let newValue = change?[NSKeyValueChangeKey.newKey] as? Array<AGSLayer>
-        let oldValue = change?[NSKeyValueChangeKey.oldKey] as? Array<AGSLayer>
+    private func changeContainsTimeAwareLayer(change: NSKeyValueObservedChange<NSMutableArray>) -> Bool {
+        let newValue = change.newValue as? Array<AGSLayer>
```

Comment:
This isn't wrong, but you could use syntactic sugar instead: `as? [AGSLayer]`.

Reply:
fixed